src/fixedrange.jl

```diff
@@ -0,0 +1,416 @@
+    transform!(Xtest, scaler)
+
+    Xscaled, scaler = fit_transform(X, FixedRangeScaler, -1, 1, obsdim=1, operate_on=[1,2,4])
+    scaler = fit_transform!(X, FixedRangeScaler, -1, 1, obsdim=1, operate_on=[1,2,4])
+
+
+
+Note that for `transform!` the data matrix `X` has to be of type <: AbstractFloat
+as the scaling occurs inplace. (E.g. cannot be of type Matrix{Int64}). This is not
+the case for `transform` however.
+For `DataFrames` `transform!` can be used on columns of type <: Integer.
+"""
+immutable FixedRangeScaler{T<:Real,U<:Real,V<:Real,W<:Real,M,I}
+    lower::T
+    upper::U
+    xmin::Vector{V}
+    xmax::Vector{W}
+    obsdim::ObsDim.Constant{M}
+    operate_on::Vector{I}
+end
+
+function FixedRangeScaler{T<:Real,N}(X::AbstractArray{T,N}; obsdim=LearnBase.default_obsdim(X), operate_on=default_scaleselection(X, convert(ObsDimension, obsdim)))
+    FixedRangeScaler(X, convert(ObsDimension, obsdim), operate_on)
+end
+
+function FixedRangeScaler{T<:Real,N,M}(X::AbstractArray{T,N}, obsdim::ObsDim.Constant{M}, operate_on)
+    xmin = vec(minimum(X, M))[operate_on]
+    xmax = vec(maximum(X, M))[operate_on]
+    FixedRangeScaler(0, 1, xmin, xmax, obsdim, operate_on)
+end
+
+function FixedRangeScaler{T<:Real,N}(X::AbstractArray{T,N}, ::ObsDim.Last, operate_on)
+    FixedRangeScaler(X, ObsDim.Constant{N}(), operate_on)
+end
+
+function FixedRangeScaler{T<:Real,N}(X::AbstractArray{T,N}, lower, upper; obsdim=LearnBase.default_obsdim(X), operate_on=default_scaleselection(X, convert(ObsDimension, obsdim)))
+    FixedRangeScaler(X, lower, upper, convert(ObsDimension, obsdim), operate_on)
+end
+
+function FixedRangeScaler{T<:Real,N,M}(X::AbstractArray{T,N}, lower, upper, obsdim::ObsDim.Constant{M}, operate_on)
+    xmin = vec(minimum(X, M))[operate_on]
+    xmax = vec(maximum(X, M))[operate_on]
+    FixedRangeScaler(lower, upper, xmin, xmax, obsdim, operate_on)
+end
+
+function FixedRangeScaler{T<:Real,N}(X::AbstractArray{T,N}, lower, upper, ::ObsDim.Last, operate_on)
+    FixedRangeScaler(X, lower, upper, ObsDim.Constant{N}(), operate_on)
+end
+
+function FixedRangeScaler(D::AbstractDataFrame; operate_on=default_scaleselection(D))
+    FixedRangeScaler(D, 0, 1, operate_on)
+end
+
+function FixedRangeScaler(D::AbstractDataFrame, lower::Real, upper::Real; operate_on=default_scaleselection(D))
+    FixedRangeScaler(D, lower, upper, operate_on)
+end
+
+function FixedRangeScaler(D::AbstractDataFrame, lower::Real, upper::Real, operate_on::AbstractVector{Symbol})
+    xmin = Float64[]
+    xmax = Float64[]
+    colnames = valid_columns(D, operate_on)
+    for colname in colnames 
+        push!(xmin, minimum(D[colname]))
+        push!(xmax, maximum(D[colname]))
+    end
+    FixedRangeScaler(lower, upper, xmin, xmax, ObsDim.Constant{1}(), colnames)
+end
+
+function StatsBase.fit{T<:Real,N}(X::AbstractArray{T,N}, ::Type{FixedRangeScaler}; obsdim=LearnBase.default_obsdim(X), operate_on=default_scaleselection(X, convert(ObsDimension, obsdim)))
```

Comment:
Yes. I changed it because of the use of `transform!()` where I think it is a Julia convention to have the variable to be modified in the first place? So `transform!(X, scaler)`. In order to have `fit`, `transform` and `fit_transform` consistent I switched the the other two to this argument order. No strong feelings about flipping all three back if that is considered the overall more concise solution.